Userland/Libraries/LibWeb/HTML/FormAssociatedElement.cpp

```diff
@@ -551,14 +551,10 @@ void FormAssociatedTextControlElement::set_the_selection_range(Optional<WebIDL::
     if (was_modified) {
         auto& html_element = form_associated_element_to_html_element();
 
-        // AD-HOC: If there is no selection, we do not fire the event. This seems to correspond to how
-        //         other browsers behave.
-        if (m_selection_start != m_selection_end) {
-            html_element.queue_an_element_task(Task::Source::UserInteraction, [&html_element] {
-                auto select_event = DOM::Event::create(html_element.realm(), EventNames::select, { .bubbles = true });
-                static_cast<DOM::EventTarget*>(&html_element)->dispatch_event(select_event);
-            });
-        }
+        html_element.queue_an_element_task(Task::Source::UserInteraction, [&html_element] {
```

Comment:
I put in the ad-hoc behavior to reflect how the select event was triggered by other browsers when the user changed the selection through the UI, not programmatically through the DOM. We should definitely retest that case for this PR :-)

Reply:
Ah true, I'd missed that

Reply:
In that case I'm unsure what behaviour we should prefer